sql/src/main/java/io/crate/planner/operators/Order.java

```diff
@@ -83,6 +88,50 @@ public LogicalPlan pruneOutputsExcept(Collection<Symbol> outputsToKeep) {
         return replaceSources(List.of(newSource));
     }
 
+    @Nullable
+    @Override
+    public FetchRewrite rewriteToFetch(Collection<Symbol> usedColumns) {
+        HashSet<Symbol> allUsedColumns = new HashSet<>(usedColumns);
+        allUsedColumns.addAll(orderBy.orderBySymbols());
+        FetchRewrite fetchRewrite = source.rewriteToFetch(allUsedColumns);
+        if (fetchRewrite == null) {
+            return null;
+        }
+        LogicalPlan newSource = fetchRewrite.newPlan();
+        Order newOrderBy = new Order(newSource, orderBy);
+        Map<Symbol, Symbol> replacedOutputs = fetchRewrite.replacedOutputs();
+        if (newOrderBy.outputs.size() > newSource.outputs().size()) {
+            // This is the case if the `orderBy` contains computations on top of the source outputs.
+            // e.g. OrderBy [x + y] where the source provides [x, y]
+            // We need to extend replacedOutputs in this case because it must always contain entries for all outputs
+            LinkedHashMap<Symbol, Symbol> newReplacedOutputs = new LinkedHashMap<>(replacedOutputs);
+            FunctionCopyVisitor<Void> mapToFetchStubs = new FunctionCopyVisitor<>() {
```

Comment:
I think we can move this in a follow up into either `fetchRewrite` itself or some utils or make it top-level, because we will need this for other operators as well.